cli/generate_operator_keys.go

```diff
@@ -5,119 +5,121 @@ import (
-					logger.Fatal("failed to parse RSA private key", zap.Error(err))
-				}
-
-				skPem := pem.EncodeToMemory(
-					&pem.Block{
-						Type:  "RSA PRIVATE KEY",
-						Bytes: x509.MarshalPKCS1PrivateKey(rsaKey),
-					},
-				)
-
-				operatorPublicKey, _ := rsaencryption.ExtractPublicKey(rsaKey)
-				publicKey, _ := base64.StdEncoding.DecodeString(operatorPublicKey)
-				sk = skPem
-				pk = publicKey
+				logger.Fatal("Failed to read private key from file", zap.Error(err))
 			}
 		}
 
 		if err := logging.SetGlobalLogger("debug", "capital", "console", nil); err != nil {
 			logger.Fatal("", zap.Error(err))
 		}
 
-		if err != nil {
-			logger.Fatal("Failed to generate operator keys", zap.Error(err))
-		}
-		logger.Info("generated public key (base64)", zap.String("pk", base64.StdEncoding.EncodeToString(pk)))
-
-		if encryptionPassword != "" {
-			encryptedData, err := keystorev4.New().Encrypt(sk, encryptionPassword)
-			if err != nil {
-				logger.Fatal("Failed to encrypt private key", zap.Error(err))
-			}
-
-			encryptedJSON, err := json.Marshal(encryptedData)
-			if err != nil {
-				logger.Fatal("Failed to marshal encrypted data to JSON", zap.Error(err))
-			}
-
-			err = os.WriteFile("encrypted_private_key.json", encryptedJSON, 0600)
-			if err != nil {
-				logger.Fatal("Failed to write encrypted private key to file", zap.Error(err))
-			}
-
-			logger.Info("private key encrypted and stored in encrypted_private_key.json")
+		if passwordFilePath != "" {
+			encryptAndSavePrivateKey(sk, passwordFilePath, logger)
 		} else {
 			logger.Info("generated public key (base64)", zap.String("pk", base64.StdEncoding.EncodeToString(pk)))
 			logger.Info("generated private key (base64)", zap.String("sk", base64.StdEncoding.EncodeToString(sk)))
 		}
 	},
 }
 
+func readPrivateKeyFromFile(filePath string) ([]byte, []byte, error) {
+	absPath, err := filepath.Abs(filePath)
+	if err != nil {
+		return nil, nil, err
+	}
+	// #nosec G304
+	keyBytes, err := os.ReadFile(absPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(string(keyBytes))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	keyPem, _ := pem.Decode(decodedBytes)
+	rsaKey, err := x509.ParsePKCS1PrivateKey(keyPem.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	skPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(rsaKey),
+	})
+
+	operatorPublicKey, err := rsaencryption.ExtractPublicKey(rsaKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	pk, err := base64.StdEncoding.DecodeString(operatorPublicKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	return skPem, pk, nil
+}
+
+func encryptAndSavePrivateKey(sk []byte, passwordFilePath string, logger *zap.Logger) {
+	absPath, err := filepath.Abs(passwordFilePath)
+	if err != nil {
+		logger.Fatal("Failed to read absolute path of password file", zap.Error(err))
+	}
+	// #nosec G304
+	passwordBytes, err := os.ReadFile(absPath)
```

Comment:
generally, I would take all I/O (file reading/writing) to different function, either the top function or just separate functions. this makes a more reusable and testable code.